great_expectations/dataset/sqlalchemy_dataset.py

```diff
@@ -230,6 +230,10 @@ def __init__(self, table_name=None, engine=None, connection_string=None,
         if self.engine.dialect.name.lower() in ["postgresql", "mysql", "sqlite", "oracle", "mssql", "oracle"]:
             # These are the officially included and supported dialects by sqlalchemy
             self.dialect = import_module("sqlalchemy.dialects." + self.engine.dialect.name)
+
+            if engine.dialect.name.lower() == "sqlite":
+                # sqlite temp tables only persist within a connection so override the engine
+                self.engine = engine.connect()
```

Comment:
This makes sense to me. I think in general we blur the line between an engine and a connection, but I suspect there are several performance-related changes we could consider based on optimizing one way or the other (i.e. this might be useful for *all* engines). I'll add it to the 'performance' bucket for the future (but think we should merge this as is).